ticktick_mcp/src/server.py

```diff
@@ -428,6 +428,515 @@ async def delete_project(project_id: str) -> str:
+    
+    if not isinstance(tasks, list):
+        return "Tasks must be provided as a list of dictionaries."
+    
+    # Validate all tasks before creating any
+    validation_errors = []
+    for i, task_data in enumerate(tasks):
+        if not isinstance(task_data, dict):
+            validation_errors.append(f"Task {i + 1}: Must be a dictionary")
+            continue
+        
+        error = _validate_task_data(task_data, i)
+        if error:
+            validation_errors.append(error)
+    
+    if validation_errors:
+        return "Validation errors found:\n" + "\n".join(validation_errors)
+    
+    # Create tasks one by one and collect results
+    created_tasks = []
+    failed_tasks = []
+    
+    try:
+        for i, task_data in enumerate(tasks):
+            try:
+                # Extract task parameters with defaults
+                title = task_data['title']
+                project_id = task_data['project_id']
+                content = task_data.get('content')
+                start_date = task_data.get('start_date')
+                due_date = task_data.get('due_date')
+                priority = task_data.get('priority', 0)
+                
+                # Create the task
+                result = ticktick.create_task(
+                    title=title,
+                    project_id=project_id,
+                    content=content,
+                    start_date=start_date,
+                    due_date=due_date,
+                    priority=priority
+                )
+                
+                if 'error' in result:
+                    failed_tasks.append(f"Task {i + 1} ('{title}'): {result['error']}")
+                else:
+                    created_tasks.append((i + 1, title, result))
+                    
+            except Exception as e:
+                failed_tasks.append(f"Task {i + 1} ('{task_data.get('title', 'Unknown')}'): {str(e)}")
+        
+        # Format the results
+        result_message = f"Batch task creation completed.\n\n"
+        result_message += f"Successfully created: {len(created_tasks)} tasks\n"
+        result_message += f"Failed: {len(failed_tasks)} tasks\n\n"
+        
+        if created_tasks:
+            result_message += "✅ Successfully Created Tasks:\n"
+            for task_num, title, task_obj in created_tasks:
+                result_message += f"{task_num}. {title} (ID: {task_obj.get('id', 'Unknown')})\n"
+            result_message += "\n"
+        
+        if failed_tasks:
+            result_message += "❌ Failed Tasks:\n"
+            for error in failed_tasks:
+                result_message += f"{error}\n"
+        
+        return result_message
+        
+    except Exception as e:
+        logger.error(f"Error in batch_create_tasks: {e}")
+        return f"Error during batch task creation: {str(e)}"
+
+# New MCP Tools for Getting things done framework (Priority / Due Dates)
+
+@mcp.tool()
```

Comment:
Note)
Separate new tools related to "Getting Things Done" framework to other file